plugin/pkg/scheduler/factory/factory.go

```diff
@@ -201,11 +208,158 @@ func NewConfigFactory(
 
 	return c
 }
 
+func (c *ConfigFactory) onPvAdd(obj interface{}) {
+	if c.enableEquivalenceClassCache {
+		pv, ok := obj.(*v1.PersistentVolume)
+		if !ok {
+			glog.Errorf("cannot convert to *v1.PersistentVolume: %v", obj)
+			return
+		}
+		c.invalidatePredicatesForPv(pv)
+	}
+}
+
+func (c *ConfigFactory) onPvDelete(obj interface{}) {
+	if c.enableEquivalenceClassCache {
+		var pv *v1.PersistentVolume
+		switch t := obj.(type) {
+		case *v1.PersistentVolume:
+			pv = t
+		case cache.DeletedFinalStateUnknown:
+			var ok bool
+			pv, ok = t.Obj.(*v1.PersistentVolume)
+			if !ok {
+				glog.Errorf("cannot convert to *v1.PersistentVolume: %v", t.Obj)
+				return
+			}
+		default:
+			glog.Errorf("cannot convert to *v1.PersistentVolume: %v", t)
+			return
+		}
+		c.invalidatePredicatesForPv(pv)
+	}
+}
+
+func (c *ConfigFactory) invalidatePredicatesForPv(pv *v1.PersistentVolume) {
+	if c.enableEquivalenceClassCache {
+		invalidPredicates := sets.NewString("MaxPDVolumeCountPredicate")
+		if pv.Spec.AWSElasticBlockStore != nil {
+			invalidPredicates.Insert("MaxEBSVolumeCount")
+		}
+		if pv.Spec.GCEPersistentDisk != nil {
+			invalidPredicates.Insert("MaxGCEPDVolumeCount")
+		}
+		if pv.Spec.AzureDisk != nil {
+			invalidPredicates.Insert("MaxAzureDiskVolumeCount")
+		}
+		c.equivalencePodCache.InvalidateCachedPredicateItemOfAllNodes(invalidPredicates)
+	}
+}
+
+func (c *ConfigFactory) onPvcAdd(obj interface{}) {
+	if c.enableEquivalenceClassCache {
+		pvc, ok := obj.(*v1.PersistentVolumeClaim)
+		if !ok {
+			glog.Errorf("cannot convert to *v1.PersistentVolumeClaim: %v", obj)
+			return
+		}
+		c.invalidatePredicatesForPvc(pvc)
+	}
+}
+
+func (c *ConfigFactory) onPvcDelete(obj interface{}) {
+	if c.enableEquivalenceClassCache {
+		var pvc *v1.PersistentVolumeClaim
+		switch t := obj.(type) {
+		case *v1.PersistentVolumeClaim:
+			pvc = t
+		case cache.DeletedFinalStateUnknown:
+			var ok bool
+			pvc, ok = t.Obj.(*v1.PersistentVolumeClaim)
+			if !ok {
+				glog.Errorf("cannot convert to *v1.PersistentVolumeClaim: %v", t.Obj)
+				return
+			}
+		default:
+			glog.Errorf("cannot convert to *v1.PersistentVolumeClaim: %v", t)
+			return
+		}
+		c.invalidatePredicatesForPvc(pvc)
+	}
+}
+
+func (c *ConfigFactory) invalidatePredicatesForPvc(pvc *v1.PersistentVolumeClaim) {
+	if c.enableEquivalenceClassCache {
+		if pvc.Spec.VolumeName != "" {
+			c.equivalencePodCache.InvalidateCachedPredicateItemOfAllNodes(sets.NewString("MaxPDVolumeCountPredicate"))
+		}
+	}
+}
+
+func (c *ConfigFactory) onServiceUpdate(oldObj interface{}, newObj interface{}) {
+	if c.enableEquivalenceClassCache {
+		// TODO(resouer) We may need to invalidate this for specified group of pods only
+		oldService := oldObj.(*v1.Service)
+		newService := newObj.(*v1.Service)
+		if !reflect.DeepEqual(oldService.Spec.Selector, newService.Spec.Selector) {
+			c.equivalencePodCache.InvalidateCachedPredicateItemOfAllNodes(sets.NewString("ServiceAffinity"))
```

Comment:
Instead of creating the sets.String on every call of this function, you can simply define high level variable:
```
var serviceAffinitySet = sets.NewString("ServiceAffinity")
```
and just use it here.

In what you are doing we will waste a loooot of memory allocations and GC time.